### PR TITLE
cmds/core/head: add stdin support

### PR DESCRIPTION
**Pull request by binjip978**

### CODE REVIEW

File: cmds/core/head/head.go [hunk 1]
```diff
@@ -36,40 +36,58 @@ func run(stdout io.Writer, stderr io.Writer, bytes, count int, files ...string)
...
-				fmt.Fprintf(stdout, "\n==> %s <==\n", filepath.Base(f.Name()))
+				fmt.Fprintf(stdout, "\n==> %s <==\n", filepath.Base(name))
 			} else {
-				fmt.Fprintf(stdout, "==> %s <==\n", filepath.Base(f.Name()))
+				fmt.Fprintf(stdout, "==> %s <==\n", filepath.Base(name))
 				newLineHeader = true
 			}
 		}
 		if printBytes {
-			n, err := io.ReadFull(f, buffer)
```
Comment (rminnich) [inline cmds/core/head/head.go:54]: We need to get read of this ReadFull. A named file can be a pipe, or a socket, or many things; or we could be reading a 10G file on a system with small memory (256MiB). We're going to need to use the buffo.NewScanner for every named file. Sorry.

**Review by rminnich — CHANGES_REQUESTED** [review]

**Comment by rminnich** [comment]
not your fault, but your stdin reading code is solid, and it's what we should use for every named file. Especially  on Plan 9, named files for IPC are common, and a ReadFull is the wrong thing to do. Apologies in advance but I suspect you can fix this quickly.

File: cmds/core/head/head.go (hunk 1 above)
Comment (binjip978) [reply cmds/core/head/head.go:54]: buffio.NewScanner is not really convenient because you need to deal with spaces, so I change buffer size to 4K

**Comment by binjip978** [comment]
> not your fault, but your stdin reading code is solid, and it's what we should use for every named file. Especially on Plan 9, named files for IPC are common, and a ReadFull is the wrong thing to do. Apologies in advance but I suspect you can fix this quickly.

here, this fault is actually all mine! :)

File: cmds/core/head/head.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ func run(stdout io.Writer, stderr io.Writer, bytes, count int, files ...string)
 	var buffer []byte
 	if bytes > 0 {
 		printBytes = true
-		buffer = make([]byte, bytes)
```
Comment (rminnich) [inline cmds/core/head/head.go:29]: please accept my apologies. My brain read "ReadFull" as "ReadFile" -- !@#$#$!

So maybe the bytes parameter is ok. Sorry :-(

File: cmds/core/head/head.go [hunk 3]
```diff
@@ -36,40 +36,68 @@ func run(stdout io.Writer, stderr io.Writer, bytes, count int, files ...string)
...
-		f, err := os.Open(file)
-		if err != nil {
-			errs = errors.Join(errs, fmt.Errorf("head: %w", err))
-			continue
-		}
+	var handle = func(r io.Reader, name string) error {
 		if len(files) > 1 {
 			if newLineHeader {
-				fmt.Fprintf(stdout, "\n==> %s <==\n", filepath.Base(f.Name()))
+				fmt.Fprintf(stdout, "\n==> %s <==\n", filepath.Base(name))
```
Comment (rminnich) [inline cmds/core/head/head.go:42]: If I do this:
head /etc/passwd /tmp/passwd
Don't I want to see the full path? I'm just wondering about reducing to filepath.Base?

File: cmds/core/head/head.go (hunk 1 above)
Comment (rminnich) [reply cmds/core/head/head.go:54]: See my note above. I was in a fog I guess.

File: cmds/core/head/head.go [hunk 4]
```diff
@@ -36,40 +36,68 @@ func run(stdout io.Writer, stderr io.Writer, bytes, count int, files ...string)
...
+					return err
+				}
+
+				stdout.Write(buffer[:min(c, n)])
+				c -= n
+				if c <= 0 {
+					break
+				}
+
+				// handle the case when user request more bytes that
```
Comment (rminnich) [inline cmds/core/head/head.go:65]: that -> than

**Review by rminnich — COMMENTED** [review]
sory about my mistake, you can change 4096 back to 'bytes'
although I'd suggest a name that's not the name of a package instead of 'bytes'

File: cmds/core/head/head.go (hunk 2 above)
Comment (binjip978) [reply cmds/core/head/head.go:29]: No worries at all. I understood your comment in following way. If user pass as bytes parameter more memory that is currently have this program will fail. And fancied such use-case  `head -c=10000000 file > f` and with 4K buffer it should work. What do you think?

File: cmds/core/head/head.go (hunk 3 above)
Comment (binjip978) [reply cmds/core/head/head.go]: done

**Review by rminnich — APPROVED** [review]

**Pull request merged**